src/main/java/edu/harvard/iq/dataverse/AbstractGlobalIdServiceBean.java

```diff
@@ -110,4 +113,18 @@ public DvObject generateIdentifier(DvObject dvObject) {
         }
         return dvObject;
     }
+    
+    /**
+     * Returns a HashMap with the same values as {@code map}. This can be either
+     * {@code map} itself, or a new instance with the same values.
+     * 
+     * This is needed as some of the internal APIs here require HashMap, but we
+     * don't want the external APIs to use an implementation class.
+     * @param <T>
+     * @param map
+     * @return A HashMap with the same values as {@code map}
+     */
+    protected <T> HashMap<T,T> asHashMap(Map<T,T> map) {
```

Comment:
This feels fishy to me. Should the internal API's really require HashMap? If so, why??

Reply:
It feels fishy, because it is fishy. I've fixed it before, but it slipped back in without me noticing during the last merge. Good catch 👍

Reply:
Oh, sorry, I take it back. this *is* the fix, which is why it is protected. The `EZIDService` class (3rd party) takes a `HashMap` as an input. So this is a generic "convert-or-case" adapter. Still a good catch, but solving this seems out of scope. Moved it to `DOIEZIdServiceBean` so it's not part of the public API, though.